packages/plugins/i18n/server/services/entity-service-decorator.js

```diff
@@ -167,18 +167,19 @@ const decorator = (service) => ({
 
     const { kind } = strapi.getModel(uid);
 
-    const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
-
-    const query = transformParamsToQuery(uid, wrappedParams);
-
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
-        return strapi.db.query(uid).findMany(query);
+        return service.findMany.call(this, uid, opts);
+      }
+      const output = service.findMany.call(this, uid, opts);
+      if (output === null || output.length === 0) {
+        return null;
       }
-      return strapi.db.query(uid).findOne(query);
+      //This could break future decorators since they don't expect the value not to be an array
+      return output[0];
     }
 
-    return strapi.db.query(uid).findMany(query);
+    return service.findMany.call(this, uid, opts);
   },
```

Comment:
By default if a CT is a Single Type, when calling findMany entityService method, it will internally call findOne.

The use case when requesting a Single Type with all it's locales becames impossible with the current Entity Service implementation.

I propose something like this:

```suggestion
    async findMany(uid, opts = {}) {
    const model = strapi.getModel(uid);

    const { isLocalizedContentType } = getService('content-types');

    if (!isLocalizedContentType(model)) {
      return service.findMany.call(this, uid, opts);
    }

    const { kind } = strapi.getModel(uid);
    const getAllLocales = opts[LOCALE_QUERY_FILTER] === 'all';

    if (kind === 'singleType' && getAllLocales) {
      return service.findMany.call(this, uid, { ...opts, ignoreKind: true });
    }

    return service.findMany.call(this, uid, opts);
  },
```

And then in the entity service findMany
```js
  async findMany(uid, opts) {
    const { kind } = strapi.getModel(uid);

    const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });

    const query = transformParamsToQuery(uid, wrappedParams);

    if (kind === 'singleType' && !opts.ignoreKind) { // <===== ignore kind
      return db.query(uid).findOne(query);
    }

    return db.query(uid).findMany(query);
  },
```

Reply:
the current implementation of the findMany in entity service already uses the find one, my proposal would fix a different thing actually
```js
  async findMany(uid, opts) {
    const { kind } = strapi.getModel(uid);

    const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });

    const query = transformParamsToQuery(uid, wrappedParams);

    if (kind === 'singleType') {
      return db.query(uid).findOne(query);
    }

    return db.query(uid).findMany(query);
  },
```

Reply:
the proposal would fix this part of the PR
```js
if (opts[LOCALE_QUERY_FILTER] === 'all') {
        return service.findMany.call(this, uid, opts);
      }
```
Which right now, it is not returning all locales of a single type, but only one (as it uses find One internally)

Reply:
mind linking me to that code?

Reply:
Ye what Marc says is the best solution but we would need to add a protection on the entity-service layer that none can passs that value